src/adapter/src/coord/message_handler.rs

```diff
@@ -72,9 +72,11 @@ impl Coordinator {
                     self.message_command(cmd).instrument(span).await
                 }
                 Message::ControllerReady => {
+                    // TODO(jkosh44) Avoid clone.
+                    let storage_metadata = self.catalog().state().storage_metadata().clone();
                     if let Some(m) = self
                         .controller
-                        .process()
+                        .process(&storage_metadata)
```

Comment:
Can someone with a bigger brain than me explain why this requires a `clone()` but no other usages of `storage_metadata()` requires a clone, even though they're passed to methods that take `mut &self` on the controller?

Reply:
Never mind, I figured it out.